### ISSUE TITLE
Opinions on a different layout

### ISSUE THREAD

**Issue by kelsin** [legitco/gimli#13]
http://javascriptplayground.com/blog/2012/04/node-js-a-todo-app-with-express/

Are we prematurely wrapping grunt around building front end files? Can we move to a system that's easier while dev'ing for a bit until we learn how to structure things?

**Comment by kelsin** [comment]
https://gist.github.com/Kelsin/eaa41c672684ffccdacc

**Comment by dotproto** [comment]
Agreed. Grunt should be limited to client builds. Server concat and magnification is an early optimization we can do without.

I'll take a crack at removing the build step for the server tonight.

**Comment by dotproto** [comment]
More broadly than using Grunt for server resources, we should discuss the layout of the project. I took a few minutes to draft out a project layout as if it were generated buy `tree`. Note that this layout is based off the project as it existed in commit 1ac418583.

```
── .env                   // Environment settings -- maintained by each indivdual developer
── .git
── .gitignore
── .travis.yml
── Gruntfile.js
── nodemon.js
── package.json
── Procfile               // Instructs heroku how to start the server
── README.md
── server.js              // Entry point to run the server application
── testem.json
├── app                   // All client deliverables
│   ├── scripts           // Client JavaScript
│   │   ├── app.js        // main entry point for client script
│   │   ├── controllers   // AngularJS controllers
│   │   ├── directives    // AngularJS directives
│   │   └── services      // AngularJS services
│   ├── styles            // Project styles -- stylus?
│   ├── images            // Static image assets
│   └── views             // AngularJS tempaltes
├── node_modules          // autogenerated by npm -install
├── server                // server resources & libs
│   ├── config
│   ├── controlers
│   ├── models
│   └── routes
├── test                  // unit, integration, and e2e tests
└── vendor                // Resources provided by 3rd parties
```

**Comment by dotproto** [comment]
Here's a second go at the same layout, but with a bit more detail.

```
── .env                   // Environment settings -- maintained by each indivdual developer
── .git
── .gitignore
── .travis.yml
── Gruntfile.js
── nodemon.js
── package.json
── Procfile               // Instructs heroku how to start the server
── README.md
── server.js              // Entry point to run the server application
── testem.json
├── client                // All client deliverables
│   ├── scripts           // Client JavaScript
│   │   ├── app.js        // main entry point for client script
│   │   ├── controllers   // AngularJS controllers
│   │   ├── directives    // AngularJS directives
│   │   └── services      // AngularJS services
│   ├── styles            // Project styles -- stylus?
│   ├── images            // Static image assets
│   ├── templates         // Common components providing base layout
│   └── views             // 'Pages' as presented by AngularJS
├── node_modules          // autogenerated by npm -install
├── server                // server resources & libs
│   ├── config
│   ├── controlers
│   ├── models
│   └── routes
├── test                  // unit, integration, etc.
│   ├── client
│   │   ├── unit          // AngularJS unit tests
│   │   └── e2e           // AngularJS end to end tests
│   ├── integration       // Tests spanning client and server
│   └── server            // Server specific tests -- may need to be defined further
└── vendor                // Resources provided by 3rd parties
```

**Comment by mschnee** [comment]
I like having the split between client/server source directories.  i'd suggest something a little more like this though:

```
── .env                   // Environment settings -- maintained by each indivdual developer
── .git
── .gitignore
── .travis.yml
── Gruntfile.js
── nodemon.js
── package.json
── Procfile               // Instructs heroku how to start the server
── README.md
── server.js              // Entry point to run the server application
── testem.json
├── node_modules          // autogenerated by npm -install
├── src
│   ├── client                // All client deliverables
│   │   ├── scripts           // Client JavaScript
│   │   │   ├── app.js        // main entry point for client script
│   │   │   ├── controllers   // AngularJS controllers
│   │   │   ├── directives    // AngularJS directives
│   │   │   └── services      // AngularJS services
│   │   ├── styles            // Project styles -- stylus?
│   │   ├── images            // Static image assets
│   │   ├── templates         // Common components providing base layout
│   │   └── views             // 'Pages' as presented by AngularJS
│   ├── server                // server resources & libs
│   │   ├── config
│   │   ├── controlers
│   │   ├── models
│   │   └── routes
│   ├── test                  // unit, integration, etc.
│   │   ├── client
│   │   │   ├── unit          // AngularJS unit tests
│   │   │   └── e2e           // AngularJS end to end tests
│   │   ├── integration       // Tests spanning client and server
│   │   └── server            // Server specific tests -- may need to be defined further
│   └── vendor                // Resources provided by 3rd parties
```

Or maybe something similar to src/{main,test}/{client,server}

**Comment by kelsin** [comment]
I could be convinced to go with a `src/{main,test}/{client,server}` but my vote is to not need it and just make `client`/`server` root and `test` root as well.

I definitely think vendor shouldn't be in src since it's not code we wrote.

**Comment by dotproto** [comment]
@mschnee The layout I proposed in my [second comment](#issuecomment-42637590) provides us with that client/server split. The primary difference between our layouts seems to be that in yours all source is grouped under `./src/` while in mine the folders are in root.

I'm going to start organizing resources as I proposed, but IMO we should continue iterating on the idea and revise file structure if necessary.
